datafusion/core/tests/fifo/mod.rs

```diff
@@ -287,6 +260,34 @@ mod unix_test {
             .await?;
         let mut stream = df.execute_stream().await?;
         let mut operations = vec![];
+
+        // Tasks
+        let mut tasks: Vec<JoinHandle<()>> = vec![];
```

Comment:
This is just code move. When error is returned during
`let mut stream = df.execute_stream().await?;`
line in the previous structure. Code stuck infinitely without returning anything. I think this is related to the where tasks are initiated. Hence changing their place, prevents this weird behaviour.